scripts/customize_conda.sh

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#            http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -euxo pipefail
+
+# This customization-script can be used to customize the conda environment.
+# It expects the following metadata:
+#   conda-component: (Required) Must be either ANACONDA or MINICONDA3.
+#   conda-env-config-uri: (Optional) Must be a gsutil path to the yaml config
+#   file.
+#   conda-packages: (Optional) A list of conda packages with versions to be
+#   installed in the base environment. Must be of the format
+#   <pkg1>:<version1>_<pkg2>:<version2>...
+#   Example: conda-packages=pytorch:1.4.0_visions:0.7.1
+#   pip-packages: (Optional) A list of pip packages with versions to be
+#   installed in the base environment. Must be of the format
+#   <pkg1>:<version1>_<pkg2>:<version2>...
+#   Example: tokenizers:0.10.1_datasets:1.5.0
+# conda-env-config-uri is mutually exclusive with conda-packages and
+# pip-packages.
+
+
+function customize_conda() {
+  local -r conda_component=$(/usr/share/google/get_metadata_value attributes/conda-component)
+  local -r conda_env_config_uri=$(/usr/share/google/get_metadata_value attributes/conda-env-config-uri)
+  local conda_packages=$(/usr/share/google/get_metadata_value attributes/conda-packages)
+  local pip_packages=$(/usr/share/google/get_metadata_value attributes/pip-packages)
+
+  validate_conda_component "${conda_component}"
+
+  if [[ -n "${conda_env_config_uri}" && (( -n "${conda_packages}" || -n "${pip_packages}" )) ]]; then
+    echo "conda-env-config-uri is mutually exclusive with conda-packages and pip-packages."
+    exit 1
+  fi
+
+  local conda_bin_dir
+  if [[ "${conda_component}" == 'ANACONDA' ]]; then
+    conda_bin_dir="/opt/conda/anaconda/bin"
+  elif [[ "${conda_component}" == 'MINICONDA3' ]]; then
+    conda_bin_dir="/opt/conda/miniconda3/bin"
+  fi
+  if [[ -n "${conda_env_config_uri}" ]]; then
```

Comment:
Consider extracting 2 functions: `customize_with_config_file` and `customize_with_package_list`.

Reply:
Is it possible to make these functions as shared scripts embedded in the image? So both Conda TLC and this custom image script call the same functions.

Reply:
It might be possible in some hacky way, but it does not add much value to us. The separators, validators and the way we are activating environment is different. The only same thing is conda and pip commands. I have refactored the code in two functions.